src/elements/toniq-pagination/toniq-pagination.element.ts

```diff
@@ -0,0 +1,152 @@
+import {assign, css, defineElementEvent, html, listen} from 'element-vir';
+import {classMap} from 'lit/directives/class-map.js';
+import {map} from 'lit/directives/map.js';
+import {pagination} from '../../augments/array';
+import {ArrowLeft24Icon, ArrowRight24Icon} from '../../icons';
+import {
+    applyBackgroundAndForeground,
+    noUserSelect,
+    removeNativeButtonStyles,
+    toniqColors,
+    toniqFontStyles,
+} from '../../styles';
+import {defineToniqElement} from '../define-toniq-element';
+import {buttonBorderRadius, ToniqButton} from '../toniq-button/toniq-button.element';
+import {ToniqIcon} from '../toniq-icon/toniq-icon.element';
+
+export const ToniqPagination = defineToniqElement({
+    tagName: 'toniq-pagination',
+    props: {
+        /** Set the selected page */
+        currentPage: 1,
+        /** This is required to set the total pages */
+        pageCount: 10,
+        /**
+         * The number of pages to show. When pageCount exceeds this number, the excess will be
+         * converted into "...". Instances of "..." are included in the pagesShown count.
+         */
+        pagesShown: 7,
+    },
+    styles: css`
+        :host {
+            display: flex;
+            align-items: center;
+            justify-content: center;
+        }
+
+        button {
+            ${removeNativeButtonStyles}
+        }
+
+        .control {
+            border-radius: ${buttonBorderRadius};
+            ${applyBackgroundAndForeground(toniqColors.accentSecondary)};
+            margin: 0 8px;
+            padding: 4px;
+        }
+
+        .control:hover {
+            ${applyBackgroundAndForeground(toniqColors.accentPrimary)};
+        }
+
+        .control[disabled] {
+            cursor: auto;
+            ${applyBackgroundAndForeground(toniqColors.accentTertiary)};
+        }
+
+        .page {
```

Comment:
Just add a `text-only` style host class to `toniq-button` and apply that instead of modifying all these styles.